elixir-oslo/crypt4gh
# Feature/multiple recipients

**Note: This does not introduce any breaking changes, current functionality is kept.**

- move test keys to tests
- support multiple recipients on encrypted filealso when decrypting file
- add linting 
- add go 1.17 build check
- other fixes

### Encrypt
```
cp test/sample.txt .
go run main.go encrypt -f sample.txt -s test/sender-C.sec.pem -p test/recipient-A.pub -p test/recipient-B.pub
go run main.go decrypt -f sample.txt.c4gh -s test/recipient-B.sec.pem
go run main.go decrypt -f sample.txt.c4gh -s test/recipient-A.sec.pe
```

### Decrypt
tested in combination with python crypt4gh
```
cp test/sample.txt .
crypt4gh encrypt --sk test/sender-C.sec.pem --recipient_pk test/recipient-A.pub --recipient_pk test/recipient-B.pub < sample.txt > sample.txt.c4gh

go run main.go decrypt -f sample.txt.c4gh -s test/recipient-B.sec.pem
go run main.go decrypt -f sample.txt.c4gh -s test/recipient-A.sec.pem
```

both decryption seems to work.

Tests will come with a follow-up PR.
